src/main/java/nova/core/component/ComponentProvider.java

```diff
@@ -60,6 +48,7 @@ public ComponentProvider() {
 	@SuppressWarnings("unchecked")
 	public <C extends ComponentMap> ComponentProvider(Class<C> componentsClass) {
 		try {
+			// TODO Figure out how to dependency injection with this without it crashing
```

Comment:
It's dependency injection that causes the crash on this line, not reflection.
It works fine like this